File: products/shopping-buddy/frontend/src/app/core/models/auth.models.ts
export type UserRole = 'Customer' | 'Buddy' | 'Admin';

export interface AuthUser {
  id: string;
  email: string;
  displayName: string;
  role: UserRole;
  roles?: UserRole[];
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface LoginResponse {
  accessToken: string;
  refreshToken: string;
  expiresIn: number;
  user: AuthUser;
}

export interface RegisterRequest {
  email: string;
  password: string;
  displayName: string;
}

export interface RegisterResponse {
  message: string;
  email: string;
  requiresVerification: boolean;
}

export interface VerifyEmailRequest {
  email: string;
  code: string;
}

export interface ResendOtpRequest {
  email: string;
  purpose: 'EmailVerification' | 'PasswordReset' | 'ForgotLoginId';
}

export interface ForgotPasswordRequest {
  email: string;
}

export interface ResetPasswordRequest {
  email: string;
  code: string;
  newPassword: string;
}

export interface ForgotLoginIdRequest {
  displayName: string;
}

export interface MessageResponse {
  message: string;
}

export interface JwtPayload {
  sub: string;
  email: string;
  name: string;
  role: UserRole;
  roles?: string;
  exp: number;
  iat: number;
}

File: products/shopping-buddy/backend/AuthApi/Models/AuthModels.cs
namespace AuthApi.Models;

public class LoginRequest
{
    public string email { get; set; } = string.Empty;
    public string password { get; set; } = string.Empty;
}

public class LoginResponse
{
    public string accessToken { get; set; } = string.Empty;
    public string refreshToken { get; set; } = string.Empty;
    public int expiresIn { get; set; }
    public AuthUser user { get; set; } = new();
}

public class AuthUser
{
    public string id { get; set; } = string.Empty;
    public string email { get; set; } = string.Empty;
    public string displayName { get; set; } = string.Empty;
    /** Primary role for legacy UI; prefer roles[] for checks. */
    public string role { get; set; } = string.Empty;
    public List<string> roles { get; set; } = new();
}

public class UserRecord
{
    public int UserId { get; set; }
    public string Email { get; set; } = string.Empty;
    public string Password { get; set; } = string.Empty;
    public string DisplayName { get; set; } = string.Empty;
    public string Role { get; set; } = string.Empty;
    public bool EmailVerified { get; set; }
    public List<string> Roles { get; set; } = new();
}

public class RegisterRequest
{
    public string email { get; set; } = string.Empty;
    public string password { get; set; } = string.Empty;
    public string displayName { get; set; } = string.Empty;
}

public class VerifyEmailRequest
{
    public string email { get; set; } = string.Empty;
    public string code { get; set; } = string.Empty;
}

public class ResendOtpRequest
{
    public string email { get; set; } = string.Empty;
    public string purpose { get; set; } = string.Empty;
}

public class ForgotPasswordRequest
{
    public string email { get; set; } = string.Empty;
}

public class ResetPasswordRequest
{
    public string email { get; set; } = string.Empty;
    public string code { get; set; } = string.Empty;
    public string newPassword { get; set; } = string.Empty;
}

public class ForgotLoginIdRequest
{
    public string displayName { get; set; } = string.Empty;
}

public class MessageResponse
{
    public string message { get; set; } = string.Empty;
}

public class RegisterResponse
{
    public string message { get; set; } = string.Empty;
    public string email { get; set; } = string.Empty;
    public bool requiresVerification { get; set; }
}
